pycbc/workflow/minifollowups.py

```diff
@@ -852,3 +1019,222 @@ def make_skipped_html(workflow, skipped_data, out_dir, tags):
     workflow += node
     files = node.output_files
     return files
+
+
+def make_upload_files(workflow, psd_files, snr_timeseries, xml_all,
+                      event_id, approximant, out_dir, channel_name,
+                      tags=None):
+    """
+    Make files including xml, skymap fits and plots for uploading to gracedb
+    for a given event
+
+    Parameters
+    ----------
+    psd_files: FileList([])
+        PSD Files from MERGE_PSDs for the search as appropriate for the
+        event
+    snr_timeseries: FileList([])
+        SNR timeseries files, one from each IFO, to add to the XML and plot
+        output from pysbs_single_template
+    xml_all: pycbc.workflow.core.File instance
+        XML file containing all events from the search
+    event_id: string
+        an integer to describe the event's position in the xml_all file
+    approximant: byte string
+        The approximant used for the template of the event, to be passed
+        to bayestar for sky location
+    out_dir:
+        The directory where all the output files should go
+    channel_name: string
+        Channel name to be added to the XML file to be uploaded
+    tags: {None, optional}
+        Tags to add to the minifollowups executables
+
+    Returns
+    -------
+    all_output_files: FileList
+        List of all output files from this process
+    """
+    indiv_xml_exe = Executable(
+        workflow.cp,
+        'generate_xml',
+        ifos=workflow.ifos, out_dir=out_dir,
+        tags=tags
+    )
```

Comment:
A more general point, with cases like this, where a particular executable is required, there should be some check that the thing in `[executables] -> generate_xml` is actually the executable that this is written against.

Reply:
That sounds a good idea, I think something like the [subclass check used for coinc_lim_for_thresh](https://github.com/gwastro/pycbc/blob/master/pycbc/events/stat.py#L153) in the stat module might be worthwhile, but that would require quite a few changes so I think is outside the scope of this PR